src/program_final.c

```diff
@@ -1 +1,348 @@
+ 	 * \details Computed support degree matrix for each of the sensors
+	 */
+    double Z[SENSOR_COUNT];
+    double totalZ=0;
+
+    
+    for(i=0;i<SENSOR_COUNT;i++) {
+        Z[i]=0;
+        for(j=i;j<SENSOR_COUNT;j++) {
+            Z[i]+=y[j]*alpha[j];
+        }
+        
+        totalZ+=Z[i];
+    }
+    
+    double prescribedrange=myAbs(totalZ/SENSOR_COUNT*1.0)*0.7;
+
+    /**
+	 * Calculate condition
+     */
+    double condZ[SENSOR_COUNT];
+    double totalCondZ=0;
+    
+    for(i=0;i<SENSOR_COUNT;i++) {
+        if(myAbs(Z[i])<prescribedrange) {
+            condZ[i]=0;
+        }
+        else {
+            condZ[i]=Z[i];
+        }
+
+        totalCondZ+=condZ[i];
+    }
+
+    /**
+	 * \brief Eliminating invalid sensor readings
+	 * \details Discard the faulty sensor readings from the support degree matrix
+	 * \details Compute weighted coefficients for each sensors from the support degree matrix
+	 */
+    double w[SENSOR_COUNT];
+
+    for(i=0;i<SENSOR_COUNT;i++) {
+        w[i] = condZ[i]/totalCondZ;
+    }
+
+	/**
+	 * \brief Compute the weight coefficient for each sensor
+	 * \details Compute the aggregated value of sensors from the weighted coefficients
+	 */
+    double fused_value=0;
+    for(i=0;i<SENSOR_COUNT;i++) {
+        fused_value+=w[i]*sensor_values[i];
+    }
+    
+    /**
+     * Return the fused_value
+     */
+    fprintf(fp,"%lf,%lf,",fused_value,prescribedrange);
+    
+    //out range sensors
+    for(i=0;i<SENSOR_COUNT;i++) {
+        if(condZ[i]==0) {
+            fprintf(fp,"sensor%d ",i+1);
+        }
+    }
+    
+    fprintf(fp,",");
+    
+    /**
+     * List of sensors that are stuck (i.e. sensors whose value does not change after a time “t”
+    */
+    for(i=0;i<SENSOR_COUNT;i++) {
+        if(sensor_values[i]==prev_sensor_values[i]) {
+            fprintf(fp,"sensor%d ",i+1);
+        }
+    }
+    fprintf(fp,"\n");
+    return ;
+}
+
+/********************************************/
+/********************************************/
+
+int main(void) {
+    char buf[1024];
+    int row_count = 0;
+
+/** 
+ * File handling operatoins are done below
+ * Input is read from .csv file from the path and opened in "r" or read mode
+ * appropriate exception handling is also done
+ * Output is generated at runtime in a .csv file format and saved at the path
+ */
+       
+    FILE *fp = fopen("../data/input.csv", "r");
+
+    if (!fp) {
+        printf("Can't open file\n");
+        return 0;
+    }
+
+    FILE *fout = fopen("../data/output.csv","w");
+    if(!fout) {
+        printf("Can't Open output file\n");
+        return 0;
+    }
+    
+    fprintf(fout,"Time(24h clock),fused value,prescribed range, outside sensors, stuck sensors\n");
+    
+    
+    char prevtime[20]="";   
+
+/**
+ * \brief Function call to initialise the reading process of each sensor
+ * \details The definition of the following two functoin calls is defined in "main.h" header file 
```

Comment:
Line should not exceed 80 Characters , Should be indented in new line
This line is about 85+ characters

Coding Convention Point 7
"Lines < 80 characters (aprox and indentation does not count towards the 80 characters) "